packages/contracts-bedrock/src/L1/SystemConfig.sol

```diff
@@ -492,6 +562,9 @@ contract SystemConfig is OwnableUpgradeable, ISemver, IGasToken {
             "SystemConfig: precision loss with target resource limit"
         );
 
+        // TODO: maxResourceLimit must be large enough to handle the SystemConfig.initialize
```

Comment:
This todo is not handled yet. We need to count the number of deposits done in system config initialize and add together their gas limits and make that the min value for max resource limit. We could also just make sure to set the resource config before the deposits are sent, which would allow resending the tx with a new max resource limit to unstick the upgrade

Reply:
Yes, thank you, that is why I have left this todo in. 

But I think it should be OK to do it in a subsequent PR (today or tomorrow) so that we can release the lock on contracts in `develop`? For the moment the buffer is on the order of 9 MM gas, so I don't think that test will impact on the implementation just yet.

Reply:
OK, here it is: 33b0792c2707